### PR TITLE
Delete user from project

### PR DESCRIPTION
**Pull request by quocquann**
## Related Tickets
[- ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/70940)

## HOW
- Delete user from project 
- Delete user form stage at the same time

### CODE REVIEW

**Comment by huongnt-2545** [comment]
commente

File: app/views.py [hunk 1]
```diff
@@ -313,3 +314,44 @@ def AddUserToProject(request, pk):
...
+        return is_in_project(user=self.request.user, project=project)
+
+    def get_queryset(self):
+        project_pk = self.kwargs.get("project_pk")
+        project = get_object_or_404(Project, pk=project_pk)
+        user_projects = (
+            UserProject.objects.filter(project=project)
+            .exclude(role=constants.PROJECT_MANAGER)
+            .order_by("role")
+        )
```
[the remark this answers is not in the document]
Comment (quocquann) [reply app/views.py:335]: chỗ này của e có 1 câu query thôi mà chị

**Comment by quocquann** [comment]
ready

**Comment by quocquann** [comment]
ready

File: app/views.py [hunk 2]
```diff
@@ -334,6 +335,27 @@ def get_context_data(self, **kwargs):
...
+    def test_func(self):
+        project_pk = self.kwargs.get("project_pk")
+        project = get_object_or_404(Project, pk=project_pk)
+        return is_in_project(user=self.request.user, project=project)
+
+    def get_queryset(self):
+        project_pk = self.kwargs.get("project_pk")
+        project = get_object_or_404(Project, pk=project_pk)
+        user_projects = UserProject.objects.filter(project=project).order_by("role")
+        UserProject.objects.select_related("user")
```
Comment (huongnt-2545) [inline app/views.py:353]:
```suggestion
        user_projects = UserProject.objects.filter(project=project).order_by("role").select_related("user")
```

File: app/views.py [hunk 3]
```diff
@@ -389,3 +411,20 @@ def add_member_to_stage(request, project_id, pk):
     }
 
     return render(request, "app/add_member_to_stage.html", context)
+
+
+def delete_member_from_project(request, project_pk, user_pk):
```
Comment (huongnt-2545) [inline app/views.py:416]: thêm require login

**Comment by quocquann** [comment]
ready

**Comment by quocquann** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]

**Pull request merged**